pkg/cloud/azure/actuators/machine_scope.go

```diff
@@ -137,6 +139,10 @@ type MachineScope struct {
 	origMachineStatus *machinev1.AzureMachineProviderStatus
 
 	machineToBePatched controllerclient.Patch
+
+	// Tags is a list of tags to apply to the resources created
+	// for the cluster
+	Tags map[string]*string
```

Comment:
Why is this pointer to string as the value? Should we not keep this the same (`map[string]string`) and transform it in the cloud library as we have been doing before?

Reply:
Tags field here is directly used for passing to the Azure APIs to add tags to the resource being created. And Azure API expects tags to be of `map[string]*string` form and is referenced at too many places. Since it's only for Azure API consumption, thought transforming at every usage instance wouldn't required and kept it in the form expected by Azure APIs.